### PR TITLE
crimson: eliminate clang build warning in crimson

### PR DESCRIPTION
**Pull request by liu-chunmei**
too many warnings when clang build.

Signed-off-by: chunmei-liu <chunmei.liu@intel.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by ronen-fr** [comment]
@liu-chunmei : nothing wrong with the (void) solution, but please take a look at the alternatives:
see https://herbsutter.com/2009/10/18/mailbag-shutting-up-compiler-warnings/

(either defining  template<class T> void ignore( const T& ) { }, which we may have already defined in
our headers and then 'ignore(x)' instead of '(void)(x)', or using the already existing
boost::ignore_unused_variable_warning())

**Comment by liu-chunmei** [comment]
> @liu-chunmei : nothing wrong with the (void) solution, but please take a look at the alternatives: see https://herbsutter.com/2009/10/18/mailbag-shutting-up-compiler-warnings/
> 
> (either defining template void ignore( const T& ) { }, which we may have already defined in our headers and then 'ignore(x)' instead of '(void)(x)', or using the already existing boost::ignore_unused_variable_warning())

Thanks Ronen, done.

File: src/crimson/tools/store_nbd/tm_driver.cc [hunk 1]
```diff
@@ -36,6 +36,7 @@ seastar::future<> TMDriver::write(
           logger().debug("dec_ref complete");
           return tm->alloc_extent<TestBlock>(t, offset, ptr.length());
         }).si_then([this, offset, &t, &ptr](auto ext) {
+	  boost::ignore_unused(offset);  //avoid clang warning;
```
Comment (rzarzynski) [inline src/crimson/tools/store_nbd/tm_driver.cc:39]: nit: oops, tab  instead of spaces.

File: src/crimson/tools/store_nbd/fs_driver.cc [hunk 2]
```diff
@@ -182,7 +182,6 @@ seastar::future<> FSDriver::mkfs()
   return init(    
   ).then([this] {
     assert(fs);
-  }).then([this] {
```
Comment (rzarzynski) [inline src/crimson/tools/store_nbd/fs_driver.cc:185]: :+1:

**Review by rzarzynski — APPROVED** [review]
LGTM with just one tiny, little nit.

**Review by ronen-fr — APPROVED** [review]

File: src/crimson/os/seastore/seastore.cc [hunk 3]
```diff
@@ -159,7 +159,9 @@ SeaStore::mount_ertr::future<> SeaStore::mount()
       oss << root << "/block." << dtype << "." << std::to_string(id);
       auto sm = std::make_unique<
 	segment_manager::block::BlockSegmentManager>(oss.str());
-      return sm->mount().safe_then([this, sm=std::move(sm), magic]() mutable {
+      return sm->mount().safe_then(
+	[this, sm=std::move(sm), magic]() mutable {
+	boost::ignore_unused(magic);  //avoid clang warning;
```
Comment (tchaikov) [inline src/crimson/os/seastore/seastore.cc:164]: might want to add a space after `//`

**Pull request merged**